bridges/ethMultiversX/bridgeExecutor.go

```diff
@@ -153,10 +155,24 @@ func (executor *bridgeExecutor) MyTurnAsLeader() bool {
 // GetBatchFromMultiversX fetches the pending batch from MultiversX
 func (executor *bridgeExecutor) GetBatchFromMultiversX(ctx context.Context) (*bridgeCore.TransferBatch, error) {
 	batch, err := executor.multiversXClient.GetPendingBatch(ctx)
-	if err == nil {
-		executor.statusHandler.SetIntMetric(core.MetricNumBatches, int(batch.ID)-1)
+	if err != nil {
+		return nil, err
+	}
+
+	executor.statusHandler.SetIntMetric(core.MetricNumBatches, int(batch.ID)-1)
+
+	isBatchInvalid := len(batch.Deposits) == 0
+	if isBatchInvalid {
+		return nil, fmt.Errorf("%w, fetched nonce: %d, num deposits: %d",
```

Comment:
is ErrFinalBatchNotFound proper error here? batch was found, but has no deposits